checker/src/org/checkerframework/checker/nondeterminism/NonDeterminismVisitor.java

```diff
@@ -54,10 +56,11 @@ public boolean isValidUse(
                                             + " collection of "
                                             + paramAnnotation),
                             tree);
+                return false;
             }
         }
 
-        return super.isValidUse(declarationType, useType, tree);
```

Comment:
Don't call super because, you've reimplement the isValidUse checker above.  (You should probably add a comment about this and about why the normal implementation doesn't work.)